{{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/events/views.py

```diff
@@ -4,18 +4,68 @@
 from .models import Event
 
 
-class EventListView(ListView):
+class BaseEventListView(ListView):
     model = Event
 
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self.featured_event = None
+
     def get_paginate_by(self, queryset):
         return self.request.pages.current.content.per_page
 
+    def get_unfiltered_queryset(self):
+        return super().get_queryset().filter(
+            page__page=self.request.pages.current,
+        )
+
     def get_queryset(self):
-        qs = super(EventListView, self).get_queryset()
+        qs = self.get_unfiltered_queryset().order_by('end_date')
+        candidates = self.get_unfiltered_queryset().order_by('-featured', 'end_date')
 
-        return qs.filter(
-            page__page=self.request.pages.current,
-        ).select_upcoming().order_by('end_date')
+        if candidates.exists():
```

Comment:
No need for this check - just use the line below this. This will add an extra query while doing nothing that the `.first()` won't (`.first()` will return `None` if no featured event is present).